src/frontend/js/delivery.js

```diff
@@ -22,6 +22,38 @@ $(function() {
         });
     });
 
+    $('.ui.order.cancel.button').click(function () {
+        var modalCtntURL = $('.ui.order.cancel.button').attr('data-url');
```

Comment:
It should be `$(this).attr()`. Otherwise it'll always return the first `data-url`.